### PR TITLE
fix(geo): adapt numbers with unit and solve show hide ditanse and area 

### PR DESCRIPTION
**Pull request by aziz-access**
**What is the current behavior?** (You can also link to an open issue here)

this behavior related to #1499 and [#1096](https://github.com/infra-geo-ouverte/igo2/issues/1096)

**What is the new behavior?**
- adapt numbers with unit changes.
- solve show hide ligne ditanse and area surface.
- keep Length and area units after navigating to another tool.

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [ ] No

**If this PR contains a breaking change, please describe the impact and migration path for existing applications:**

**Other information**:

### CODE REVIEW

File: packages/geo/src/lib/measure/measurer/measurer.component.ts [hunk 1]
```diff
@@ -527,7 +523,12 @@ export class MeasurerComponent implements OnInit, OnDestroy {
    */
   onAreaUnitChange(unit: MeasureAreaUnit) {
     this.activeAreaUnit = unit;
-    this.table.refresh();
```
Comment (alecarn) [inline packages/geo/src/lib/measure/measurer/measurer.component.ts:530]: @aziz-access J'ai supprimé le `table.refresh()`. En théorie, avec le changement que tu as fait on ne devrait plus avoir besoin de l'invoquer. Good for you?

**Review by alecarn — APPROVED** [review]

**Comment by alecarn** [comment]
@pelord ready for your review

File: packages/geo/src/lib/measure/measurer/measurer.component.ts (hunk 1 above)
Comment (aziz-access) [reply packages/geo/src/lib/measure/measurer/measurer.component.ts:530]: ok good

**Review by alecarn — APPROVED** [review]

**Review by pelord — APPROVED** [review]

**Review by pelord — CHANGES_REQUESTED** [review]
We need to keep the defined units check this video on tool change 
chrome_2023-11-27_11-58-08

**Review by pelord — APPROVED** [review]

**Pull request merged**